### PR TITLE
Write blog post to announce JVM language support

### PR DESCRIPTION
**Pull request by olafurpg**

### CODE REVIEW

File: blogposts/2021/java-scala-kotlin-code-intelligence.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+title: 'Precise code intelligence for Java, Scala, and Kotlin'
+description: 'We are excited to announce that Sourcegraph now supports precise code intelligence for Java, Scala, and Kotlin.'
+author: Ólafur Páll Geirsson
+publishDate: 2021-12-17T00:00-07:00
+tags: [blog]
+slug: java-scala-kotlin-code-intelligence
+heroImage: //images.ctfassets.net/le3mxztn6yoo/t4Qpcq5kA0AYM24Ws4mOk/4edf5502a936bbec90c262fa00355aed/sourcegraph-mark.png
+socialImage: https://about.sourcegraph.com/blog/blogIndexing1200x627.jpg
```
Comment (olafurpg) [inline blogposts/2021/java-scala-kotlin-code-intelligence.md:9]: I copy-pasted this image from another blog post but we should probably have something else. Would it make sense to use the cross-repo example instead?

Comment (nickmyyz) [reply blogposts/2021/java-scala-kotlin-code-intelligence.md:9]: I think that would make sense!

**Review by nickmyyz — APPROVED** [review]
I think this looks great! Ready to go on my end :)

**Comment by rebeccadee** [comment]
@olafurpg Is this ready to publish? Nick is on PTO now so please let me know if there's anything I can do to get this out 😃

**Comment by olafurpg** [comment]
Sounds good, we can publish today.

> One question for you: this was mentioned in relation to log4j and supporting users with their remediation efforts but I don't see that highlighted anywhere. Is there a way to bring that in (as appropriate)?

While I'd love to mention log4j somehow I think it's a stretch to call it out today. I'm confident precise code intel will play a significant role in the future for incidents like this but lsif-java didn't play any part in our log4j efforts so far.

**Comment by rebeccadee** [comment]
@olafurpg 

> While I'd love to mention log4j somehow I think it's a stretch to call it out today. I'm confident precise code intel will play a significant role in the future for incidents like this but lsif-java didn't play any part in our log4j efforts so far.

Thanks for explaining! We can leave it out. I will give this a once-over before merging 🎉

File: blogposts/2021/java-scala-kotlin-code-intelligence.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+publishDate: 2021-12-17T00:00-07:00
+tags: [blog]
+slug: java-scala-kotlin-code-intelligence
+heroImage: //images.ctfassets.net/le3mxztn6yoo/t4Qpcq5kA0AYM24Ws4mOk/4edf5502a936bbec90c262fa00355aed/sourcegraph-mark.png
+socialImage: https://about.sourcegraph.com/blog/java-cross-language-example.png
+published: true
+---
+
+We are excited to announce that Sourcegraph now supports precise code intelligence for Java, Scala, and Kotlin.
+More precisely, this means you can get compiler-accurate “Go to definition” and “Find references” within a git repository and all transitive dependencies of your codebase. Dependencies can come from any Maven repository, such as Maven Central, Artifactory or Sonatype Nexus. The versions Java 8, Java 11, Java 17, Scala 2, Scala 3, and Kotlin 1.x are supported.
```
Comment (rebeccadee) [inline blogposts/2021/java-scala-kotlin-code-intelligence.md:14]:
```suggestion
More precisely, this means you can get compiler-accurate “Go to definition” and “Find references” within a Git repository and all transitive dependencies of your codebase. Dependencies can come from any Maven repository, such as Maven Central, Artifactory or Sonatype Nexus. The versions Java 8, Java 11, Java 17, Scala 2, Scala 3, and Kotlin 1.x are supported.
```

File: blogposts/2021/java-scala-kotlin-code-intelligence.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+Cross-repository navigation on Sourcegraph makes it possible to find usages of symbols across thousands of repositories–unlike a local IDE that only shows usages in your own project. For example, in the image below, we find usages of the `java.util.Iterator.forEachRemaining()` method in the repositories apache/kafka and quarkus/quarkus ([try it yourself](https://sourcegraph.com/jdk@v11/-/blob/java.base/java/util/Iterator.java?L130:18#tab=references)).
+
+![](/blog/java-cross-repo-example.png)
+
+Cross-language navigation between Java, Scala, and Kotlin languages works as you would expect: You can find Kotlin or Scala usages from Java code and vice-versa. For example, below we find usages of `java.util.concurrent.atomic.AtomicBoolean.get()` in a Kotlin file from the kotest/kotest repo and get Scala examples in the apache/kafka repository ([try it yourself](https://sourcegraph.com/github.com/kotest/kotest@7ccb6ef/-/blob/kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/atomic/AtomicBooleanMatchers.kt?L77:40#tab=references)).
```
Comment (rebeccadee) [inline blogposts/2021/java-scala-kotlin-code-intelligence.md:20]:
```suggestion
Cross-language navigation between Java, Scala, and Kotlin languages works as you would expect: You can find Kotlin or Scala usages from Java code and vice versa. For example, below we find usages of `java.util.concurrent.atomic.AtomicBoolean.get()` in a Kotlin file from the kotest/kotest repo and get Scala examples in the apache/kafka repository ([try it yourself](https://sourcegraph.com/github.com/kotest/kotest@7ccb6ef/-/blob/kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/atomic/AtomicBooleanMatchers.kt?L77:40#tab=references)).
```

File: blogposts/2021/java-scala-kotlin-code-intelligence.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+![](/blog/java-cross-language-example.png)
+
+For Gradle, Maven, and sbt users, the easiest way to get started is to run the `sourcegraph/lsif-java` Docker container.
+
+```sh
+docker run -v $(pwd):/home/gradle sourcegraph/lsif-java lsif-java index
+src lsif upload
+```
+
+For optimal performance, we recommend using the Java-based launcher with a pre-populated build cache to minimize the overhead from downloading dependencies. Check out the [lsif-java website](https://sourcegraph.github.io/lsif-java) for more detailed installation instructions. Note that it’s possible to integrate lsif-java with other build tools like Bazel but doing so requires manual configuration.
```
Comment (rebeccadee) [inline blogposts/2021/java-scala-kotlin-code-intelligence.md:31]:
```suggestion
For optimal performance, we recommend using the Java-based launcher with a pre-populated build cache to minimize the overhead from downloading dependencies. Check out the [lsif-java website](https://sourcegraph.github.io/lsif-java) for more detailed installation instructions. Note that it’s possible to integrate lsif-java with other build tools like Bazel, but doing so requires manual configuration.
```

**Comment by scalabilitysolved** [comment]
Looks good, excited to see it go out! :rocket:

**Comment by macraig** [comment]
LGTM! 🚢

File: blogposts/2021/java-scala-kotlin-code-intelligence.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+---
+title: 'Precise code intelligence for Java, Scala, and Kotlin'
+description: 'We are excited to announce that Sourcegraph now supports precise code intelligence for Java, Scala, and Kotlin.'
+author: Ólafur Páll Geirsson
+publishDate: 2021-12-17T00:00-07:00
```
Comment (rebeccadee) [inline blogposts/2021/java-scala-kotlin-code-intelligence.md:5]:
```suggestion
publishDate: 2021-12-20T00:00-07:00
```

File: blogposts/2021/java-scala-kotlin-code-intelligence.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
+---
+title: 'Precise code intelligence for Java, Scala, and Kotlin'
+description: 'We are excited to announce that Sourcegraph now supports precise code intelligence for Java, Scala, and Kotlin.'
```
Comment (rebeccadee) [inline blogposts/2021/java-scala-kotlin-code-intelligence.md:3]:
```suggestion
description: 'We are excited to announce that Sourcegraph now supports precise code intelligence for Java, Scala, and Kotlin, enabling compiler-accurate “Go to definition” and “Find references” within a Git repository and all transitive dependencies of your codebase.'
```

**Review by rebeccadee — COMMENTED** [review]
Apply suggestions

File: blogposts/2021/java-scala-kotlin-code-intelligence.md (hunk 6 above)
Comment (rebeccadee) [reply blogposts/2021/java-scala-kotlin-code-intelligence.md]: @olafurpg I didn't apply this change as I want to check the language makes sense. I wanted to add some more detail to the description so it's not as repetitive of the title

**Comment by olafurpg** [comment]
@rebeccadee Thank you! Those were great suggestions.

File: blogposts/2021/java-scala-kotlin-code-intelligence.md (hunk 6 above)
Comment (olafurpg) [reply blogposts/2021/java-scala-kotlin-code-intelligence.md]: It's great to have a longer description. The blog post was looking pretty empty from the `/blog/` overview page 
image

**Pull request merged**